cmd/query/app/static_handler.go

```diff
@@ -94,11 +122,64 @@ func NewStaticAssetsHandler(staticAssetsRoot string, options StaticAssetsHandler
 		}
 		indexBytes = basePathPattern.ReplaceAll(indexBytes, []byte(fmt.Sprintf(basePathReplace, options.BasePath)))
 	}
-	return &StaticAssetsHandler{
-		options:   options,
-		indexHTML: indexBytes,
-		assetsFS:  assetsFS,
-	}, nil
+
+	return indexBytes, nil
+}
+
+func (sH *StaticAssetsHandler) watch() {
+	if sH.options.UIConfigPath == "" {
+		return
+	}
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		sH.options.Logger.Error("failed to create a new watcher for the UI config", zap.Error(err))
+		return
+	}
+
+	go func() {
+		for {
+			select {
+			case event := <-watcher.Events:
+				if event.Op&fsnotify.Remove == fsnotify.Remove {
+					// this might be related to a file inside the dir, so, just log a warn if this is about the file we care about
+					// otherwise, just ignore the event
+					if event.Name == sH.options.UIConfigPath {
+						sH.options.Logger.Warn("the UI config file has been removed, using the last known version")
+					}
+					continue
+				}
+
+				// this will catch events for all files inside the same directory, which is OK if we don't have many changes
+				sH.options.Logger.Info("reloading UI config", zap.String("filename", sH.options.UIConfigPath))
+
+				content, err := loadIndexBytes(sH.assetsFS.Open, sH.options)
+				if err != nil {
+					sH.options.Logger.Error("error while reloading the UI config", zap.Error(err))
+				}
+
+				sH.indexHTML.Store(content)
+			case err, ok := <-watcher.Errors:
+				if !ok {
+					return
+				}
+				sH.options.Logger.Error("event", zap.Error(err))
+			}
+		}
+	}()
+
+	err = watcher.Add(sH.options.UIConfigPath)
+	if err != nil {
+		sH.options.Logger.Error("error adding watcher to file", zap.String("file", sH.options.UIConfigPath), zap.Error(err))
+	}
+	sH.options.Logger.Info("watching", zap.String("file", sH.options.UIConfigPath))
```

Comment:
should this go under `else`?

Reply:
Good catch. I had a `return` in a previous iteration, but it's now gone, so, it does need to be within an else statement.